en/core-api/e-wallet.md

```diff
@@ -302,49 +351,73 @@ You will get the **API response** like the following.
+</article>
+</details>
 
-### Show QR Code Image (on computer or tablet)
+### 2. Altering payment flow depending on the device used
 
-To display transaction QR Code image, use the url from `generate-qr-code` actions retrieved from API response. Simplest way is to **"hotlink"** the image url, if the frontend is HTML, put the url in image tag `<img src="[QR CODE URL]">`, or display it on a similar component without downloading.
-If the frontend does not support such scenario, download the QR code image from that url, then display it on frontend.
+The user flow is different for a web browser (on a computer or a tablet) and a smartphone.
 
-Instruction Example for **QR Code** :
+#### Show QR Code Image (on computer or tablet)
 
-1. Tap **Pay using GoPay**
-2. QR code will appear on the next page
-3. Open any **QRIS compatible app** (e.g: Gojek) on your mobile phone
-4. Tap **Pay** then scan the QR Code
-5. Check and verify your payment details then tap **PAY**
-6. Complete your security PIN / verification
-7. Your transaction is finished
+To display transaction QR Code image, use the URL from `generate-qr-code` actions retrieved from API response. Simplest way is to **"hotlink"** the image URL. If the frontend is HTML, put the URL in image tag `<img src="[QR CODE URL]">`, or display it on a similar component without downloading.
+If the frontend does not support such scenario, download the QR code image from that URL, then display it on frontend.
+
+Instruction Example for **QR Code** : 
+
+1. Select *Payment Mode*, for example **Pay using GoPay**.
+2. Click **Pay Now**. QR code will be displayed.
+3. Open any **QRIS compatible app**, installed in your phone, for example Gojek.
+4. Scan the **QR code**.
+5. Click **Pay**, after checking and verifying your payment details.
+6. Verify your **Security PIN** and finish your transaction.
+
+   
 
 ![GoPay QR Instruction](./../../asset/image/core-api_gopay-qr-pay.png)
 
-### Create Redirect Link to Gojek Apps (on SmartPhone)
+#### Creating Redirect Link to Gojek Apps (on smartphone)
 
-To redirect customer to Gojek app, use the url from `deeplink-redirect` actions retrieved from API response. Then customer can be redirected via server-side redirect, using javascript like `window.location=[DEEPLINK URL]`, or using HTML link `<a href="[DEEPLINK URL]">Pay with GoPay</a>`.
+To redirect the customer to Gojek app, use URL from `deeplink-redirect` actions retrieved from API response. Then customer can be redirected via server-side redirect, using JavaScript, such as `window.location=[DEEPLINK URL]`, or using HTML link, `<a href="[DEEPLINK URL]">Pay with GoPay</a>`.
 
-Instruction Example for **Deeplink** :
+The steps for **Deeplink** are as given below:
 
-1. Tap **Pay using GoPay**
-2. You will be redirected to **Gojek** app
-3. Check and verify your payment details then tap **Pay**
-4. Enter your security **PIN**
-5. Your transaction is finished
+1. Select *Payment mode*, for example **Pay using GoPay**.
```

Comment:
Your replacement for step 1 is changing the meaning. Customer should literally do the `Tap **Pay using GoPay**`, it cannot be replaced by `Select ...`. Please fix